app/contracts/relations/base_contract.rb

```diff
@@ -55,6 +55,13 @@ def validate_to_exists
     end
 
     def validate_nodes_relatable
+      # when creating a relation from the work package relations tab and not selecting a WorkPackage
+      # the to_id is not set
+      # in this case we only want to show the "WorkPackage can't be blank" error instead of a
+      # misleading circular dependencies error
+      # the error is added by the models presence validation
+      return if model.to_id.nil?
```

Comment:
This is probably the most critical part of that PR. Can that have unwanted side effects?